src/builder/value_parser.rs

```diff
@@ -936,15 +906,108 @@ impl Default for OsStringValueParser {
 /// Implementation for [`ValueParser::path_buf`]
 ///
 /// Useful for composing new [`TypedValueParser`]s
-#[derive(Copy, Clone, Debug)]
+#[derive(Clone)]
 #[non_exhaustive]
-pub struct PathBufValueParser {}
+pub struct PathBufValueParser {
+    canonicalize: bool,
+    starts_with: Option<String>,
+    ends_with: Option<String>,
```

Comment:
I'm not seeing these defined in click.  Let's hold off on these in this PR and instead discuss their use case in the issue